pkg/controller/batchrelease/batchrelease_plan_executor.go

```diff
@@ -124,38 +123,38 @@ func (r *Executor) executeBatchReleasePlan(release *v1alpha1.BatchRelease, newSt
 		switch {
```

Comment:
When progressDone is true, i think it do not come into Finalizing, but rather just to Completed. Finalizing State feel like it can be removed.

Reply:
Finalizing state cannot be skipped.
Finalizing state will delete some resources/annotations and restore some configuration for workloads.